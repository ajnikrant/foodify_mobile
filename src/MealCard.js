import React, {useState, useEffect} from 'react';
import { useHistory } from 'react-router-dom'


function MealCard({meal, allergenSearch}){
    const {description, id, image, ingredients, name, price} = meal
    const [cardColor, setCardColor] = useState(false)
    const [showDescription, setShowDescription] = useState(false)

    const history = useHistory()


    function handleCardClick(){
        history.push(`/meals/${id}`, { params: meal })

    }

   
    function highlightAllergens(){

        if (allergenSearch === ""){
            setCardColor(false)
        }

        else if (ingredients.toLowerCase().includes(allergenSearch.toLowerCase())){
                setCardColor(true)
        }
        else {setCardColor(false)}
    }

    useEffect(()=> {
        highlightAllergens()
    }, [allergenSearch])    

    function handleHoverEnter(){
        setShowDescription(true)
    }

    function handleHoverLeave(){
        setShowDescription(false)
    }

    return(
        <div className={cardColor? "mealcard card text-white bg-danger mb-3" : "mealcard card text-dark bg-light mb-3"} style={{ width: '18rem' }} 
        onMouseLeave={handleHoverLeave} 
        onMouseEnter={handleHoverEnter} 
        onClick={handleCardClick}
        >
            <h4>{name}</h4>
            <p>{showDescription ? 
                <div className="mouseOverDesc">
                    <p>{description}</p>
                    <img className="mouseOverDescImage" src = {image} alt={description}/>
                </div> 
                : 
                <img className="menuCardImage" src = {image} alt={description}/>}
            </p>
            <p id="price"> ${price}</p>
            

        </div>
    )

}

export default MealCard
