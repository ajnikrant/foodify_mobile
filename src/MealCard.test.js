import React, { createElement } from 'react';
import ReactDom from 'react-dom';
import { render } from '@testing-library/react';
import MealCard from './MealCard';

const mockMeal = {
    name: "Test Meal",
    image: "someURL",
    price: 1, 
    description: "test desc",
    ingredients: "test ingredients"
  };

  const mockSearchEntry = "test"
  
  // **************** Test one ****************
test('Meal Card component matches snapshot', () => {
    const { asFragment } = render(<MealCard meal={mockMeal} allergenSearch={mockSearchEntry} />);

    expect(asFragment()).toMatchSnapshot();
});

// **************** Test two ****************
test('Card renders without crashing', () => {

    const div = document.createElement('div');

    render(
        <MealCard meal={mockMeal} allergenSearch={mockSearchEntry} />, div
    );
});

