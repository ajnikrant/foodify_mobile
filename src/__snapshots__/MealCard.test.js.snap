// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal Card component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mealcard card text-white bg-danger mb-3"
    style="width: 18rem;"
  >
    <h4>
      Test Meal
    </h4>
    <p>
      <img
        alt="test desc"
        class="menuCardImage"
        src="someURL"
      />
    </p>
    <p
      id="price"
    >
       $1
    </p>
  </div>
</DocumentFragment>
`;
